### PR TITLE
Add an optional flag to activate a mission config during upload

### LINKED ISSUE #14: Add an optional flag to `/zeus-upload` to activate the mission immediately
Would save a separate call to `/zeus-set-mission`

### PR DESCRIPTION
**Pull request by Gehock**
Fixes #14. Split across several commits, but can be squashed.

I'm *slightly* hesitant to add the new flag to `zeus_upload()`, because activating a config isn't really the job of the upload function. But again, I don't want to make that a responsibility of the Discord glue either, and an `upload_and_potentially_activate()` wouldn't really make sense to me either.

### CODE REVIEW

File: src/zeusops_bot/cogs/zeus_upload.py [hunk 1]
```diff
@@ -46,12 +46,18 @@ def __init__(self, bot: "ZeusopsBot", reforger_confgen: ReforgerConfigGenerator)
         input_type=discord.SlashCommandOptionType.attachment,
         required=False,
     )
+    @discord.option(
+        "activate",
+        description="Automatically set the uploaded mission as the active config",
```
Comment (OverkillGuy) [inline src/zeusops_bot/cogs/zeus_upload.py]:
```suggestion
        description="Immediately use this uploaded mission as the active mission",
```

File: features/reforger_upload.feature [hunk 2]
```diff
@@ -14,6 +14,14 @@ Scenario: Upload next mission
   Then a new server config file is created
   And the config file is patched with <modlist.json> and <scenarioId>
 
+Scenario: Upload next mission and activate
+  Given a Zeusops mission locally ready
+  When Zeus calls "/zeus-upload"
+  And Zeus specifies <modlist.json>, <scenarioId>, <filename>, <activate>
```
Comment (OverkillGuy) [inline features/reforger_upload.feature]: Potentially here we only care about the activate flag, as the rest is implied to be "used normally"?
In the Gherkin we don't have to be exhaustive, just make the meaning of the feature clear.

```suggestion
  When Zeus calls "/zeus-upload" with the activate flag
```

File: src/zeusops_bot/reforger_config_gen.py [hunk 3]
```diff
@@ -63,6 +66,8 @@ def zeus_upload(
         target_filepath = as_config_file(self.target_dest, filename)
         # Create the file itself
         target_filepath.write_text(json.dumps(modded_config_dict, indent=4))
+        if activate:
+            self.zeus_set_mission(target_filepath)
```
Comment (OverkillGuy) [inline src/zeusops_bot/reforger_config_gen.py:75]: Feels weird to be doing THAT here, isn't it more of a standalone action call in the discord bot itself?
The discord bot, taking the activate flag, calls these 2 "services" one after the other" instead?

Or would you rather see the "activate" as a legit option of the upload feature? I can be tempted

File: tests/test_upload_mission.py [hunk 4]
```diff
@@ -24,14 +27,20 @@ def test_upload_edits_files(base_config: Path, mission_dir: Path):
...
     config = json.loads(out_path.read_text())
     assert config["game"]["scenarioId"] == scenario_id, "Should update scenarioId"
     assert isinstance(config["game"]["mods"], list)
     assert config["game"]["mods"][0] == MODLIST_DICT[0]
+    if activate:
+        # And the server config file is set as the active mission
+        target = mission_dir / "current-config.json"
+        assert target.readlink() == out_path.relative_to(
+            mission_dir
+        ), "Target should point to uploaded file"
```
Comment (OverkillGuy) [inline tests/test_upload_mission.py]: That feels very weird: we're deduping by matrixing it, but then using one-half via if? awkwaaaaard!

Suggest instead keeping the full test as is, just to see if anything breaks of the normal feature when activating or not (and even then, it's not like activating is super important for the normal feature).
And separately have a more dedicated test that just tests the upload does whatever (other tests cover it) AND ACTIVATE specifically.

Effectively suggesting that  we split this into its own new test, the current one being overlaoded too much

**Review by OverkillGuy — CHANGES_REQUESTED** [review]
Weird test, not sure yet about the set-mission being not in the discord side either

File: tests/test_upload_mission.py (hunk 4 above)
Comment (Gehock) [reply tests/test_upload_mission.py]: The reason why I combined the upload and activation in one test is that testing the activation separately would require duplicating basically all of the upload test code as well, since the configs have to be set up anyway. Not really sure what would be the best way of implementing that without copy-pasting large blocks of code

File: src/zeusops_bot/reforger_config_gen.py (hunk 3 above)
Comment (Gehock) [reply src/zeusops_bot/reforger_config_gen.py:75]: It is a bit of a strange location for activating the mission, I agree. But I'd also want to avoid doing anything extra in the Discord glue, because it cannot be tested easily. So ideally I'd want to keep the Discord command handlers strictly in the format of
1. call a method that contains the "actual functionality"
2. handle possible raised exceptions
3. report success/failure.

Anything beyond that starts to become hidden functionality that isn't tested in any way.

File: tests/test_upload_mission.py (hunk 4 above)
Comment (Gehock) [reply tests/test_upload_mission.py]: As far as I can see, there would be a couple of potential options how to split the tests, but I'm not very happy about either of them.

## Option 1
```py
def test_upload_edits_files(base_config: Path, mission_dir: Path):
    """Scenario: Upload next mission creates file"""
    # Given a Zeusops mission locally ready
    # And Zeus specifies <modlist.json>, <scenarioId>, <filename>
    scenario_id = "cool-scenario-1"
    filename = "Jib_20250228"
    config_gen = ReforgerConfigGenerator(
        base_config_file=base_config, target_folder=mission_dir
    )
    # When Zeus calls "/zeus-upload"
    modlist = extract_mods(MODLIST_JSON)
    out_path = config_gen.zeus_upload(scenario_id, filename, modlist, activate=False)
    # Then a new server config file is created
    assert out_path.is_file(), "Should have generated a file on disk"
    # And the config file is patched with <modlist.json> and <scenarioId>
    config = json.loads(out_path.read_text())
    assert config["game"]["scenarioId"] == scenario_id, "Should update scenarioId"
    assert isinstance(config["game"]["mods"], list)
    assert config["game"]["mods"][0] == MODLIST_DICT[0]


def test_upload_activate_mission(base_config: Path, mission_dir: Path):
    """Scenario: Upload next mission and activate"""
    # Given a Zeusops mission locally ready
    # When Zeus calls "/zeus-upload" with the activate flag
    scenario_id = "cool-scenario-1"
    filename = "Jib_20250228"
    config_gen = ReforgerConfigGenerator(
        base_config_file=base_config, target_folder=mission_dir
    )
    # When Zeus calls "/zeus-upload"
    modlist = extract_mods(MODLIST_JSON)
    out_path = config_gen.zeus_upload(scenario_id, filename, modlist, activate=True)
    # Then a new server config file is created
    assert out_path.is_file(), "Should have generated a file on disk"
    # And the config file is patched with <modlist.json> and <scenarioId>
    config = json.loads(out_path.read_text())
    assert config["game"]["scenarioId"] == scenario_id, "Should update scenarioId"
    assert isinstance(config["game"]["mods"], list)
    assert config["game"]["mods"][0] == MODLIST_DICT[0]
    # And the server config file is set as the active mission
    target = mission_dir / "current-config.json"
    assert target.readlink() == out_path.relative_to(
        mission_dir
    ), "Target should point to uploaded file"
```
This is just the original test split into two functions based on the `activate` flag. Performs mostly the same tests in both functions and contains a large amount of duplicated code. Breaks the DRY principle massively and makes it very easy for the test functions to start to diverge if changes are made in the common code in either of them.

## Option 2
```py
def test_upload_edits_files(base_config: Path, mission_dir: Path):
    # (same content as in Option 1)


def test_upload_activate_mission(base_config: Path, mission_dir: Path):
    """Scenario: Upload next mission and activate"""
    # Given a Zeusops mission locally ready
    # When Zeus calls "/zeus-upload" with the activate flag
    config_gen = ReforgerConfigGenerator(
        base_config_file=base_config, target_folder=mission_dir
    )
    # When Zeus calls "/zeus-upload"
    modlist = extract_mods(MODLIST_JSON)
    out_path = config_gen.zeus_upload("cool-scenario-1", "Jib_20250228", modlist, activate=True)
    # Then the server config file is set as the active mission
    target = mission_dir / "current-config.json"
    assert target.readlink() == out_path.relative_to(
        mission_dir
    ), "Target should point to uploaded file"
```
Less duplicated code, only performs the basic mission set-up in order to test the symlink. In the unlikely case that the `activate` flag somehow breaks mission patching or other aspects of uploading while still creating the symlink correctly, the error **will not be caught**.

Comment (Gehock) [reply tests/test_upload_mission.py]: I guess Option 3 would be to somehow call the first test as part of the second test, that would provide more exhaustive test coverage while keeping the duplicated code to a minimum.

Comment (OverkillGuy) [reply tests/test_upload_mission.py]: AS discussed, solution 2 but with first testcase keeping the parametrize of activate flag is nice

File: src/zeusops_bot/reforger_config_gen.py (hunk 3 above)
Comment (OverkillGuy) [reply src/zeusops_bot/reforger_config_gen.py:75]: As discussed, let's take it as is and move on =)

**Review by OverkillGuy — APPROVED** [review]
Just gotta parametrize the option 2 and we're good
